api-registration/server/index.js

```diff
@@ -0,0 +1,68 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+
+  argon2.hash(password)
+    .then(hashedPw => {
+      const sql = `
+    INSERT INTO "users" ("username", "hashedPassword")
+    VALUES ($1, $2)
+    RETURNING *
+  `;
+      const params = [username, hashedPw];
+      db.query(sql, params)
+        .then(result => {
+          const [user] = result.rows;
+          if (user) {
+            res.status(201).json(user);
```

Comment:
You don't want to send the password back to the client, even if it is hashed. The best way to control the data being sent back to the client is to specify the desired columns in the `returning` clause of the SQL statement.
```SQL
returning "userId", "username", "createdAt"
```

Reply:
Oh yeah... woops. 
![image](https://user-images.githubusercontent.com/75236087/192685398-8cde330c-3311-4503-acf7-202b962966e3.png)